### PR TITLE
Message queue delete op - Use correct class for DI when instantiating ProductUpdateService

### PR DESCRIPTION
**Pull request by supercid**

### CODE REVIEW

**Comment by supercid** [comment]
Don't merge this yet, I'll check after my vacation 🌴 .
There's still a bug when fetching what has actually been deleted vs updated
Prolly somewhere here https://github.com/Nosto/nosto-magento2/blob/develop/Model/Indexer/ProductIndexer.php#L148
Might not be the correct logic altogether to fetch what has been removed. 🤔 
image

**Review by dairbuirabass — APPROVED** [review]

**Pull request merged**
